### PR TITLE
pin matplotlib to <3.4 in 4.2.x

### PR DESCRIPTION
**Pull request by eteq**
This is for the 4.2.x branch, as a potential change for a 4.2.1post1 just to get the RTD build working.

### CODE REVIEW

**Comment by pllim** [comment]
The `samp` failure is fixed by #11447 -- FYI

**Comment by pllim** [comment]
RTD passed and the failures thus far are known issues and non-blocking... 🤞

File: setup.cfg
Comment (pllim) [inline setup.cfg:87]: When I tried, just pinning it here wasn't enough because `all` overwrites it. I had to pin it like in #11462 .

**Review by pllim — COMMENTED** [review]
I guess this is good enough for a bugfix branch, unless you prefer something else like #11466 (if it works).

**Comment by pllim** [comment]
Eh,.... RTD passed! Maybe I am going crazy.

**Comment by pllim** [comment]
Ah,,, because we skipped the `!=3.4.0` pin here.

**Review by pllim — APPROVED** [review]
I am approving to remove the block. I'll leave it up to you as the release manager if this is good enough or you want a solution that is consistent with `master` (not sure what yet).

**Comment by eteq** [comment]
I made #11468 because I think @nstarman's solution is better if it works!  But if that doesn't work in 4.2.x out-of-the-box it's not worth the trouble right now, and instead we should go with this one.

**Comment by pllim** [comment]
Closed without merge in favor of #11468

**Pull request closed without merging**
